frontend/src/components/problem/ProblemDisplay.tsx

```diff
@@ -71,6 +87,24 @@ function ProblemDisplay(props: ProblemDisplayParams) {
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState('');
 
+  const onDragEnd = (result: any) => {
+    // dropped outside the list
+    if (!result.destination) {
+      return;
+    }
+
+    const newTestCases: TestCase[] = reorder(
+      newProblem.testCases,
+      result.source.index,
+      result.destination.index,
+    );
+
+    const updatedProblem: Problem = newProblem;
+    updatedProblem.testCases = newTestCases;
+
+    setNewProblem(updatedProblem);
```

Comment:
Potential alternative based on [this post](https://stackoverflow.com/questions/37755997/why-cant-i-directly-modify-a-components-state-really) on directly modifying React state (btw I didn't test this suggestion so some of the syntax might be wrong):

```suggestion
    setNewProblem({ ...newProblem, testCases: newTestCases });
```